sem-install

```diff
@@ -85,15 +83,52 @@ install::install_php() {
   return $?
 }
 
+install::install_elixir() {
+  response1=$(curl --write-out %{http_code} --head --silent --output /dev/null $(install::package_url)/${language}/elixirs_${language_version}.tar.gz)
+  response2=$(curl --write-out %{http_code} --head --silent --output /dev/null $(install::package_url)/${language}/archives_${language_version}.tar.gz)
+  if [[ $response1 -ne 200 ]] || [[ $response2 -ne 200 ]]; then
+    case ${language_version} in
```

Comment:
If I put `case` outside then each case will have to have `if-else` branching with repeated code in it (allthough it can be shortened with functions).
To me it looks like it's easier just to copy-paste the same `case` after the `else` (before it starts to download elixir packages). This way if appropriate erlang is not there it will be installed first and then elixir download and install will continue.

Reply:
@bogyo210 so you'll want to extract this `case` block to a separate function and then use it, or modify it so that it can be more generic?

Reply:
@skipi 
I thought at movig the `case` block before the `if`, so erlang is activated (downloaded), for both branches of the `if-else`. It would be easier to maintain and more error-proof since adding elixir-erlang versions needs just one modification.

Reply:
I got confused, I understood that I should wrap `case` around `if-else`, but now I realize that I can put it before `if` like this:

```
install::install_elixir() {
  response1=$(curl --write-out %{http_code} --head --silent --output /dev/null $(install::package_url)/${language}/elixirs_${language_version}.tar.gz)
  response2=$(curl --write-out %{http_code} --head --silent --output /dev/null $(install::package_url)/${language}/archives_${language_version}.tar.gz)
  case ${language_version} in
      "1.7"* | "1.8"*)
        sem-version erlang 21.3
        ;;
      "1.9"* | "1.10"*)
        sem-version erlang 22.3
        ;;
      "1.11"*)
        sem-version erlang 23.3
        ;;
      "1.12"*)
        sem-version erlang 24.0
        ;;

      *)
        echo "Building of '${language}' version '${language_version}' is not supported"
        return 1;
        ;;
  esac
  if [[ $response1 -ne 200 ]] || [[ $response2 -ne 200 ]]; then
    kiex install ${language_version}
    kiex use ${language_version}
    mix local.hex --force
  else
    curl --silent $(install::package_url)/${language}/elixirs_${language_version}.tar.gz --output elixirs_${language_version}.tar.gz
    curl --silent $(install::package_url)/${language}/archives_${language_version}.tar.gz --output archives_${language_version}.tar.gz
    tar -zxf elixirs_${language_version}.tar.gz && rm -rf elixirs_${language_version}.tar.gz
    tar -zxf archives_${language_version}.tar.gz && rm -rf archives_${language_version}.tar.gz
    mv elixirs/* ~/.kiex/elixirs/
    mv archives/* ~/.kiex/mix/archives/
    kiex use ${language_version}
  fi
  return $?
}
```

Reply:
:shipit: LGTM